src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -964,14 +964,18 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	private async updateKernelInfoOnKernelChange(kernel: nb.IKernel, kernelAlias?: string) {
 		await this.updateKernelInfo(kernel);
 		kernelAlias = this.kernelAliases.find(kernel => this._defaultLanguageInfo?.name === kernel.toLowerCase()) ?? kernelAlias;
```

Comment:
Shouldn't the kernelAlias be set to undefined here when switching from Kusto to SQL?

Reply:
If the kernel is currently Kusto, the defaultLanguageInfo.name would be kusto, and as such the kernelAlias would also be Kusto. The kernelspec would be SQL, that's why we need to check for the selectedKernelDisplayName so that we know when a user is changing the kernels to set the kernelAlias to undefined.